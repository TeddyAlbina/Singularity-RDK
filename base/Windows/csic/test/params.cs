public class test {
	public int this[params string[] args] {
		get { return 1; }
		set { print(args); System.Console.WriteLine("value={0}", value); }
	}
	static void print(params string[] args) {
		if (args.Length == 0)
			System.Console.WriteLine("no arguments");
		else
			foreach (string s in args)
				System.Console.Write(s);
	}
	static public void Main() {
		print("Hello", " ", "World", "\n");
		print(new string[] {"Hello", " ", "World", "\n"});
		test x = new test();
		x["Hello", " ", "World", "\n"] = 2;
		print();
	}
}