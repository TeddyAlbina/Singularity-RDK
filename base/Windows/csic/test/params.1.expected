Hello World
Hello World
Hello World
value=2
no arguments
